source/exe/signal_action.cc

```diff
@@ -0,0 +1,119 @@
+#include "exe/signal_action.h"
+
+#include <signal.h>
+#include <sys/mman.h>
+
+#include "common/common/assert.h"
+
+constexpr int SignalAction::FATAL_SIGS[];
+
+void SignalAction::SigHandler(int sig, siginfo_t* info, void* context) {
+  void* error_pc = 0;
+
+  const ucontext_t* ucontext = reinterpret_cast<const ucontext_t*>(context);
+  if (ucontext != nullptr) {
+#ifdef REG_RIP
+    // x86_64
+    error_pc = reinterpret_cast<void*>(ucontext->uc_mcontext.gregs[REG_RIP]);
+#elif defined(REG_EIP)
+// x86 Classic - not tested
+// error_pc = reinterpret_cast<void*>(ucontext->uc_mcontext.gregs[REG_EIP]);
+#warning "Please enable and test x86_32 pc retrieval code in signal_action.cc"
+#elif defined(__arm__)
+// ARM - not tested
+// error_pc = reinterpret_cast<void*>(ucontext->uc_mcontext.arm_pc);
+#warning "Please enable and test ARM pc retrieval code in signal_action.cc"
+#elif defined(__ppc__)
+// PPC - not tested
+// error_pc = reinterpret_cast<void*>(ucontext->uc_mcontext.regs->nip);
+#warning "Please enable and test PPC pc retrieval code in signal_action.cc"
+#else
+#warning "Cannot determine PC location in machine context for your architecture"
+#endif
+  }
+
+  BackwardsTrace tracer(true);
+  tracer.LogFault(strsignal(sig), info->si_addr);
+  if (error_pc != 0) {
+    tracer.CaptureFrom(error_pc);
+  } else {
+    tracer.Capture();
+  }
+  tracer.Log();
+
+  signal(sig, SIG_DFL);
+  raise(sig);
+}
+
+void SignalAction::InstallSigHandlers() {
+  stack_t stack;
+  stack.ss_sp = altstack_ + GUARD_SIZE; // Guard page at one end ...
+  stack.ss_size = ALTSTACK_SIZE;        // ... guard page at the other
+  stack.ss_flags = 0;
+
+  if (sigaltstack(&stack, nullptr) < 0) {
```

Comment:
nit: Just drop the if statement here and just RELEASE_ASSERT the call directly, same for all cases below. You don't need any of the error output. Line # is enough.